src/common/test-webserver.c

```diff
@@ -34,17 +34,38 @@ typedef struct {
   CockpitWebServer *web_server;
   gchar *localport;
   gchar *hostport;
+
+  const gchar *expected_protocol;
+  gchar *expected_remote;
```

Comment:
It feels wrong to put expectations into the fixture. This should be the actual state, i. e. just `protocol` and `remote`? Expectiations should live in the TestCase only.

Reply:
The issue here is that (particularly in the case of `expected_remote`) there is quite a lot of logic in the fixture setup function to decide what these values will eventually become.  And in the end, we need to have a single pointer to pass through to the verification function.

I could indeed essentially perform all of that logic, statically, and move it to the testcase structs and pass those through instead.  Thoughts?

Reply:
I tried to do this and it's really not possible without a significant rethink here, or duplication of some code: there's no way to know the remote address in advance, because for some test cases it's the IP address of the machine that the test is running on.

Reply:
Ack, nevermind then. Thanks for checking!